quinn-h3/benches/bench.rs

```diff
@@ -106,77 +119,41 @@ impl Context {
-            let _enter = my_span.enter();
             let mut runtime = rt();
-            let handle = runtime.spawn(
-                async move {
-                    let mut incoming = server.build().unwrap();
-                    let mut incoming_req = incoming
-                        .next()
-                        .await
-                        .expect("accept")
-                        .await
-                        .expect("connect");
-                    debug!("recv incoming");
-                    while let Some(recv_req) = incoming_req.next().await {
-                        let (request, mut body_reader, sender) = recv_req.await.expect("recv_req");
-                        while let Some(_) = body_reader.data().await {}
-
-                        let mut body_writer = sender
-                            .send_response(
-                                Response::builder().status(StatusCode::OK).body(()).unwrap(),
-                            )
-                            .await
-                            .expect("send_response");
-
-                        let frame_size = request
-                            .headers()
-                            .get("frame_size")
-                            .map(|x| x.to_str().unwrap().parse().expect("parse frame size"))
-                            .expect("no frame size");
-                        let mut remaining = request
-                            .headers()
-                            .get("total_size")
-                            .map(|x| x.to_str().unwrap().parse().expect("parse total size"))
-                            .expect("no total size");
-
-                        let data = "a".repeat(frame_size);
-                        while remaining > 0 {
-                            let size = std::cmp::min(frame_size, remaining);
-                            body_writer
-                                .write_all(&data.as_bytes()[..size])
-                                .await
-                                .expect("body write");
-                            remaining -= size;
-                        }
-                    }
-                }
-                .instrument(error_span!("server")),
-            );
+            let handle = runtime.spawn(async {
+                let incoming_conn = server.with_socket(socket).unwrap();
+                handle_connection(incoming_conn, stop_recv).instrument(error_span!("server"))
+            });
             runtime.block_on(handle).unwrap();
         });
 
         (addr, handle)
     }
 
+    pub fn stop_server(&mut self) {
+        if let Some(send) = self.stop_server.take() {
+            send.send(()).expect("stop server");
+        }
+    }
+
     pub fn make_client(&self, server_addr: SocketAddr) -> (client::Connection, Runtime) {
         let mut runtime = rt();
         let my_span = span!(Level::TRACE, "client");
         let _enter = my_span.enter();
+        dbg!(server_addr);
```

Comment:
Probably shouldn't leave this in?

Reply:
Yeah, plus it's not that recommended for a benchmark to print in terminal :)